### PR TITLE
Update Go pdk docs

### PR DESCRIPTION
**Pull request by gszr**
### Summary

This PR mentions the deprecation of the go-pluginserver -- aka "Legacy
Style". The External Plugins guide is updated so that it consistently advises
uses to use the embedded pluginserver option for Go.

### Reason

The external go-pluginserver is deprecated and scheduled to be removed in 2.8. As such, docs need to advise users to use the new embedded pluginserver option for Go plugins.

### CODE REVIEW

**Comment by gszr** [comment]
Requested review from @javierguerragiraldez as SME.

**Comment by gszr** [comment]
I figured it would be useful to have a section clearly defining what steps are needed to update from "legacy" to new style. Will add that shortly.

**Review by javierguerragiraldez — APPROVED** [review]

**Comment by gszr** [comment]
This is now ready for review.

File: app/gateway/2.8.x/reference/external-plugins.md [hunk 1]
```diff
@@ -66,8 +67,11 @@ plugins = bundled, go-hello, js-hello, py-hello
 plugin server using a different configuration style. Starting with {{site.base_gateway}} version 2.3,
 the old style is recognized and internally transformed to the new style.
 
+**Note**: the legacy configuration is deprecated in {{site.base_gateway}} version 2.8.0, to be
+removed in {{site.base_gateway}} version 3.0.0.
```
Comment (lena-larionova) [inline app/gateway/2.8.x/reference/external-plugins.md:71]:
```suggestion
{:.important}
> The legacy configuration is deprecated in {{site.base_gateway}} version 2.8.0 and 
> planned to be removed in {{site.base_gateway}} version 3.0.0.
```

File: app/gateway/2.8.x/reference/external-plugins.md [hunk 2]
```diff
@@ -76,17 +80,30 @@ Property | Description | Default
 
 Notes:
 
-- The old style doesn't allow multiple plugin servers.
+- Usage of the legacy style is discouraged; it is deprecated as of {{site.base_gateway}} version 2.8.0, to be
+removed in {{site.base_gateway}} version 3.0.0.
```
Comment (lena-larionova) [inline app/gateway/2.8.x/reference/external-plugins.md:84]:
```suggestion
- Using the legacy style is discouraged; it is deprecated as of {{site.base_gateway}} version 2.8.0, 
and planned to be removed in {{site.base_gateway}} version 3.0.0.
```

File: app/gateway/2.8.x/reference/external-plugins.md [hunk 3]
```diff
@@ -76,17 +80,30 @@ Property | Description | Default
...
+- The legacy configuration doesn't allow multiple plugin servers.
 - Version 0.5.0 of [go-pluginserver] requires the old style configuration.
-- The new style configuration requires v0.6.0 of [go-pluginserver]
+- The new style configuration requires v0.6.0 of [go-pluginserver].
```
Comment (lena-larionova) [inline app/gateway/2.8.x/reference/external-plugins.md:87]:
```suggestion
- The new configuration style requires v0.6.0 of [go-pluginserver].
```

File: app/gateway/2.8.x/reference/external-plugins.md [hunk 4]
```diff
@@ -76,17 +80,30 @@ Property | Description | Default
...
 
-## Developing Go plugins
+#### Updating from legacy to embedded server style
+
+The update from the legacy configuration to embedded plugin server style encompasses
+the following steps:
```
Comment (lena-larionova) [inline app/gateway/2.8.x/reference/external-plugins.md:92]:
```suggestion
Update legacy configuration to embedded plugin server style configuration with
the following steps:
```

File: app/gateway/2.8.x/reference/external-plugins.md [hunk 5]
```diff
@@ -76,17 +80,30 @@ Property | Description | Default
...
 
-{{site.base_gateway}} support for the Go language consist of two parts:
+1. Add a `main()` function that calls `server.StartServer(New, Version, Priority)`.
+2. Ensure that properties `go_plugins_dir` and `go_pluginserver_exe` are not set
+in your Kong configuration file or environment variable.
+3. Set configuration according to [Kong Gateway plugin server configuration](#kong-gateway-plugin-server-configuration).
+Note that, in this mode, the plugin itself acts as the plugin server; as such
+the external go-pluginserver is no longer required.
```
Comment (lena-larionova) [inline app/gateway/2.8.x/reference/external-plugins.md:99]:
```suggestion
```
Moving this to the start of the section.

File: app/gateway/2.8.x/reference/external-plugins.md [hunk 6]
```diff
@@ -76,17 +80,30 @@ Property | Description | Default
...
+- Usage of the legacy style is discouraged; it is deprecated as of {{site.base_gateway}} version 2.8.0, to be
+removed in {{site.base_gateway}} version 3.0.0.
+- The legacy configuration doesn't allow multiple plugin servers.
 - Version 0.5.0 of [go-pluginserver] requires the old style configuration.
-- The new style configuration requires v0.6.0 of [go-pluginserver]
+- The new style configuration requires v0.6.0 of [go-pluginserver].
 
-## Developing Go plugins
+#### Updating from legacy to embedded server style
+
```
Comment (lena-larionova) [inline app/gateway/2.8.x/reference/external-plugins.md:90]:
```suggestion
#### Updating from legacy to embedded server style

In embedded server mode, the plugin itself acts as the plugin server, 
so the external `go-pluginserver` is no longer required.

```

File: app/gateway/2.8.x/reference/external-plugins.md [hunk 7]
```diff
@@ -76,17 +80,30 @@ Property | Description | Default
...
+2. Ensure that properties `go_plugins_dir` and `go_pluginserver_exe` are not set
+in your Kong configuration file or environment variable.
+3. Set configuration according to [Kong Gateway plugin server configuration](#kong-gateway-plugin-server-configuration).
+Note that, in this mode, the plugin itself acts as the plugin server; as such
+the external go-pluginserver is no longer required.
 
-- [go-pdk] as a library, provides Go functions to access {{site.base_gateway}} features of the [PDK][kong-pdk].
-- [go-pluginserver] an executable to dynamically load plugins written in Go.
+Check out the [go-plugins](https://github.com/Kong/go-plugins/tree/v0.5.0) repository for an example of the required updates; plugins with the `-lm` suffix correspond to the legacy method, while those without the suffix
+correspond to the embedded plugin server approach.
```
Comment (lena-larionova) [inline app/gateway/2.8.x/reference/external-plugins.md:102]:
```suggestion
Check out the [go-plugins](https://github.com/Kong/go-plugins/tree/v0.5.0) repository for an example of the required updates. Plugins with the `-lm` suffix correspond to the legacy method, while those without the suffix
correspond to the embedded plugin server approach.
```

File: app/gateway/2.8.x/reference/external-plugins.md [hunk 8]
```diff
@@ -76,17 +80,30 @@ Property | Description | Default
...
+
+## Developing Go plugins
 
+{{site.base_gateway}} support for the Go language consist of the [go-pdk], a library that provides Go functions to access {{site.base_gateway}} features of the [PDK][kong-pdk].
```
Comment (lena-larionova) [inline app/gateway/2.8.x/reference/external-plugins.md:106]:
```suggestion
{{site.base_gateway}} supports the Go language with the [go-pdk], a library that provides Go functions to access {{site.base_gateway}} features of the [PDK][kong-pdk].
```

File: app/gateway/2.8.x/reference/external-plugins.md [hunk 9]
```diff
@@ -164,7 +167,45 @@ You can add more fields to the structure and they’ll be passed around, but
...
+the imports list, and add a `main()` function:
+
+```go
+func main () {
+  server.StartServer(New, Version, Priority)
+}
+```
+
+Note that the `main()` function must have a `package main` line at the
+top of the file.
```
Comment (lena-larionova) [inline app/gateway/2.8.x/reference/external-plugins.md:182]:
```suggestion
The `main()` function must have a `package main` line at the
top of the file.
```

File: app/gateway/2.8.x/reference/external-plugins.md [hunk 10]
```diff
@@ -164,7 +167,45 @@ You can add more fields to the structure and they’ll be passed around, but
...
+
+Then, a standard Go build creates an executable. There is no extra go-pluginserver,
```
Comment (lena-larionova) [inline app/gateway/2.8.x/reference/external-plugins.md:184]:
```suggestion
Then, a standard Go build creates an executable. There is no extra `go-pluginserver`,
```

File: app/gateway/2.8.x/reference/external-plugins.md [hunk 11]
```diff
@@ -164,7 +167,45 @@ You can add more fields to the structure and they’ll be passed around, but
...
+no plugin loading, and no compiler/library/environment compatibility issues.
```
Comment (lena-larionova) [inline app/gateway/2.8.x/reference/external-plugins.md:185]:
```suggestion
no plugin loading, and no compiler, library, or environment compatibility issues.
```

**Review by lena-larionova — COMMENTED** [review]
Minor feedback, thanks for the updates!

**Review by lena-larionova — APPROVED** [review]
LGTM

**Pull request merged**